### ISSUE TITLE
FlutterError thrown on payment method creation on Android

### ISSUE THREAD

**Issue by lbk9** [flutter-stripe/flutter_stripe#203]
**Describe the bug**
The UI behaves as expected, the user is able to interact with the CardField. However, when I press the associated button with the cardfield and call
`await Stripe.instance
        .createPaymentMethod(PaymentMethodParams.card())
        .then((paymentMethod) {
      paymentMethod = paymentMethod;
      _handlePayment(paymentMethod);
    }).catchError(setError);`

A FlutterError is thrown

**To Reproduce**
1. Fill out card details
2. Attempt to create payment method
```
FlutterError (setState() called after dispose(): _PlatformViewLinkState#7e735(lifecycle state: defunct, not mounted)
This error happens if you call setState() on a State object for a widget that no longer appears in the widget tree (e.g., whose parent widget no longer includes the widget in its build). This error can occur when code calls setState() from a timer or an animation callback.
The preferred solution is to cancel the timer or stop listening to the animation in the dispose() callback. Another solution is to check the "mounted" property of this object before calling setState() to ensure the object is still in the tree.
This error might indicate a memory leak if setState() is being called because another object is retaining a reference to this State object after it has been removed from the tree. To avoid memory leaks, consider breaking the reference to this object during dispose().)
```

**Expected behavior**

**Smartphone / tablet**
 - Device: OnePlus 9 Pro
 - OS: Android 11
 - Package version: 1.1.0
 - Flutter version 2.2.3

**Additional context**

**Comment by jonasbark** [comment]
Please post a minimal project where this is reproducible.
The error you posted indicates that it's a developer error, unrelated to the plugin itself

**Comment by ericel** [comment]
Doesn't work in ios either. The card payment method is kinda broken. There are a lot of issues here on this method but no solution so far!!

**Comment by jonasbark** [comment]
Then help us by providing a reproducible example.

**Comment by lbk9** [comment]
I've attempted to create a reproducible example using my with MobX as I have in the project where I am running into. The issue is not showing here so it must be something on my side.

I'm not sure what the issue is on my side but the issue only ever occurs on the specific page and when the CardField widget when I have entered payment details and attempt to complete an API call to confirm the paymentMethod. No other point in the app does it and it has only began to happen when I have implemented the Stripe package.
